roxie/topo/toposerver.cpp

```diff
@@ -299,13 +302,47 @@ int main(int argc, const char *argv[])
         else
             globals->loadProp(argv[i], true);  // We ignore unrecognized options for now
     }
-    traceLevel = globals->getPropInt("--traceLevel", 1);
-    topoPort = globals->getPropInt("--port", TOPO_SERVER_PORT);
-    if (globals->getPropBool("--stdlog", traceLevel != 0))
+
+    // locate settings xml file in runtime dir
+    char currentDirectory[_MAX_DIR];
+    if (!getcwd(currentDirectory, sizeof(currentDirectory)))
+    {
+        perror("getcwd failure");
+        return EXIT_FAILURE;
+    }
+    topologyFile.set(currentDirectory);
+    addNonEmptyPathSepChar(topologyFile);
+    topologyFile.append(PATHSEPCHAR).append("toposerver.xml");
+    IPropertyTree *topology;
+    if (checkFileExists(topologyFile.str()))
+        topology = createPTreeFromXMLFile(topologyFile.str(), ipt_lowmem);
+    else
+        topology = createPTreeFromXMLString(
+            "<TopoServerProcess port='9004' traceLevel='1'>"
+            "</TopoServerProcess>"
+            , ipt_lowmem
+            );
+    if (globals->hasProp("--traceLevel"))
```

Comment:
@richardkchapman this section will properly overwrite anything in the settings file, with anything supplied on the command line. Just tested and it's properly assigning to whatever port I give, and writing out logs to whichever new directory I give it.